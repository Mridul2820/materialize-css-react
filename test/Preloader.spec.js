import React from 'react';
import { render } from '@testing-library/react';
import Preloader from '../src/Preloader';

describe('<Preloader />', () => {
  test('should render a Preloader', () => {
    const { container } = render(
      <Preloader
        className="find"
        size="big"
        active={false}
        color="red"
        flashing
      />
    );

    expect(container).toMatchSnapshot();
  });

  test('should render a Preloader with default properties', () => {
    const { container } = render(<Preloader />);

    expect(container).toMatchSnapshot();
  });
});
