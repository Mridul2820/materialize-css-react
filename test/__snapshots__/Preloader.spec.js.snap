// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preloader /> should render a Preloader 1`] = `
<div>
  <div
    class="find preloader-wrapper big"
  >
    <div
      class="spinner-layer spinner-blue"
    >
      <div
        class="circle-clipper left"
      >
        <div
          class="circle"
        />
      </div>
      <div
        class="gap-patch"
      >
        <div
          class="circle"
        />
      </div>
      <div
        class="circle-clipper right"
      >
        <div
          class="circle"
        />
      </div>
    </div>
    <div
      class="spinner-layer spinner-red"
    >
      <div
        class="circle-clipper left"
      >
        <div
          class="circle"
        />
      </div>
      <div
        class="gap-patch"
      >
        <div
          class="circle"
        />
      </div>
      <div
        class="circle-clipper right"
      >
        <div
          class="circle"
        />
      </div>
    </div>
    <div
      class="spinner-layer spinner-yellow"
    >
      <div
        class="circle-clipper left"
      >
        <div
          class="circle"
        />
      </div>
      <div
        class="gap-patch"
      >
        <div
          class="circle"
        />
      </div>
      <div
        class="circle-clipper right"
      >
        <div
          class="circle"
        />
      </div>
    </div>
    <div
      class="spinner-layer spinner-green"
    >
      <div
        class="circle-clipper left"
      >
        <div
          class="circle"
        />
      </div>
      <div
        class="gap-patch"
      >
        <div
          class="circle"
        />
      </div>
      <div
        class="circle-clipper right"
      >
        <div
          class="circle"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Preloader /> should render a Preloader with default properties 1`] = `
<div>
  <div
    class="preloader-wrapper active"
  >
    <div
      class="spinner-layer spinner-blue-only"
    >
      <div
        class="circle-clipper left"
      >
        <div
          class="circle"
        />
      </div>
      <div
        class="gap-patch"
      >
        <div
          class="circle"
        />
      </div>
      <div
        class="circle-clipper right"
      >
        <div
          class="circle"
        />
      </div>
    </div>
  </div>
</div>
`;
